gadget/mountedfilesystem.go

```diff
@@ -299,6 +307,41 @@ func fsStructBackupPath(backupDir string, ps *LaidOutStructure) string {
 	return filepath.Join(backupDir, fmt.Sprintf("struct-%v", ps.Index))
 }
 
+func maybePreserveManagedBootAssets(mountPoint string, ps *LaidOutStructure) ([]string, error) {
+	if ps.Role != SystemSeed && ps.Role != SystemBoot {
+		return nil, nil
+	}
+
+	opts := &bootloader.Options{
```

Comment:
I think I understand this but I think it warrants a comment. (I'm talking about the Recovery and NoSlashBoot options set below).

Reply:
Added a comment